### ISSUE TITLE
xmap doesn't support jvp on top of it

### ISSUE THREAD

**Issue by bloops** [jax-ml/jax#7957]
Calling jvp on an `xmap`ped function gives a NotImplementedError.

```python
import jax
import jax.numpy as jnp
from jax import vmap
from jax.experimental.maps import xmap

def f(r, x):
  return r * x * x

xs = jnp.arange(4.)
vmf = lambda r: vmap(partial(f, r), 0, 0)(xs)
xmf = lambda r: xmap(partial(f, r), [0], [0])(xs)

print(vmf(2.)) # works. Prints [ 0.  2.  8. 18.]
print(xmf(2.)) # works. Prints [ 0.  2.  8. 18.]

print(jax.jvp(vmf, (2.,), (1.,))) # works. ([ 0.,  2.,  8., 18.], [0., 1., 4., 9.])
print(jax.jvp(xmf, (2.,), (1.,))) # NotImplementedError!
```

**Comment by apaszke** [comment]
Just for context: xmap implements standard JVP just fine (via `ad.JVPTrace.process_call`), but it doesn't implement the `post_process` rule which is necessary to handle this case correctly. cc @mattjj  who wanted to get rid of post processing?

**tl;dr** To work around this until it's fixed you can make `r` an explicit input to `xmap` which you then ignore. You can use `None` for its `in_axes` and `out_axes.

**Comment by bloops** [comment]
Thanks! The workaround works for the toy example.

(As a relatively minor aside, could we have xmap accept a float instead of insisting on jnp.array? It complains with this stack trace if I pass in a float like `xmf(2.)`)
 
```python
google3/third_party/py/jax/experimental/maps.py in fun_mapped(*args)
    607     if has_input_rank_assertions:
    608       for arg, spec in zip(args_flat, in_axes_flat):
--> 609         if spec.expected_rank is not None and spec.expected_rank != arg.ndim:
    610           raise ValueError(f"xmap argument has an in_axes specification of {spec.user_repr}, "
    611                            f"which asserts that it should be of rank {spec.expected_rank}, "

AttributeError: 'float' object has no attribute 'ndim'
```

Here is the fixed toy example which now works.
```python
def f(r, x):
  return r * x * x

xs = jnp.arange(4.)
vmf = lambda r: vmap(partial(f, r), 0, 0)(xs)
xmf = lambda r: xmap(f, ([], [0]), [0])(r, xs)

print(vmf(2.)) # works. Prints [ 0.  2.  8. 18.]
print(xmf(jnp.array(2.))) # works. Prints [ 0.  2.  8. 18.]
print(jax.jvp(vmf, (2.,), (1.,))) # works. ([ 0.,  2.,  8., 18.], [0., 1., 4., 9.])
print(jax.jvp(xmf, (2.,), (1.,))) # works! ([ 0.,  2.,  8., 18.], [0., 1., 4., 9.])
```

**Comment by bloops** [comment]
In order to address this issue, I see that for JVPTrace `process_xmap`, it's just a passthrough to `ad.JVPTrace.process_call`
https://github.com/google/jax/blob/b992034eafb3f9800ad986760d60ba8c57563fc5/jax/experimental/maps.py#L860

Can the post_process similarly be a passthrough to `ad.JVPTrace`? Like so? (Seems to work for this toy example, although I have no idea if this is the correct approach!)
```
class XMapPrimitive(core.MapPrimitive):
  # ...
  def post_process(self, trace, out_tracers, params):
    return trace.post_process_xmap(self, out_tracers, params)
# ...
ad.JVPTrace.post_process_xmap = ad.JVPTrace.post_process_call
```

**Comment by mattjj** [comment]
Just noticed my cc here! (I need to improve my inbox organization for that kind of @-mention stuff...)

Indeed I think we can get rid of post_process stuff this year.

@bloops that fix sounds plausible to me, though I haven't looked at the code! The logic should be pretty much the same, I think.

**Comment by bloops** [comment]
Thanks @mattjj!

@apaszke does that sound good for a short term fix? If so, I can send the minimal PR with just that one-line change and the toy-example as the unit test.

**Comment by mattjj** [comment]
Thank you as always, @rajasekharporeddy !
